recipes/opengl/all/conanfile.py

```diff
@@ -0,0 +1,52 @@
+from conans import ConanFile, tools
+from conans.errors import ConanException
+import os
+
+
+class SysConfigOpenGLConan(ConanFile):
+    name = "opengl"
+    version = "system"
```

Comment:
In the Bincrafters' version I did go with `virtual` as it also has the option to use MESA drivers via Conan as opt-in. We can't move Mesa just yet to CCI, so we can't add this option right now, but it would be good to leave the opportunity for later in a non-breaking or confusing way.

What do you think changing the version name to `virtual`?

Also pinging for @ericLemanissier's opinion on this

Reply:
+1 for virtual

Reply:
there is an alternate option for opt-in as we have discussed.
you may easily define an override like `opengl/vendor`.
or `self.requires("opengl/vendor", override=True)` on the recipe level.
so, still we have a fine-grain control to choose an implementation.

Reply:
@czoido @danimtb @uilianries @jgsogo  
what do you think on naming `opengl/system` vs `opengl/virtual`?
for me it's not a big deal, both are fine

Reply:
I think `system` says something about what it is behind: this recipe will install (opt-in) things in my system and populate a cpp_info to these libraries from my system.

Reply:
`opengl/vendor` does not allow to specify versions, which mesa has (see https://github.com/bincrafters/conan-mesa/blob/testing/20.0.1/conandata.yml)

Reply:
I think that is a different issue. If we had packages for all the libraries, what would you like to do in your consumer library?
 * will you require `opengl` and this reference will be like a wrapper/facade that will require different libraries? based on options? based on the operating system?
 * will you implement conditions in your recipe and based on options you will require different references?

Reply:
the first seems more maintainable, as it has a unique point choosing the opengl implementation, so it avoids having different implementations in a dependency graph. this is what @Croydon started to do in https://github.com/bincrafters/conan-opengl/blob/testing/virtual/conanfile.py#L62

Reply:
so, maybe, we need here for the `opengl` _graph_ more recipes:
 * `opengl/virtual`? that will require different implementations/references based on options, probably these options should have different defaults for different OSs. Do we have any other example for a wrapper/facade recipe?
 * and then we need a bunch of `xxxxx/system` recipes to wrap system-installed `mesa`, `gl`, `glvnd`,... then the user could override `mesa/my-version` if needed.

Does it make sense? Is this what we want?

Reply:
Hi guys! Do you ever considered using OpenGL API version as virtual package version? I mean `opengl/4.5` for the [spec version 4.5](https://www.khronos.org/registry/OpenGL/specs/gl/glspec45.core.pdf) for example.

In fact, what we mean when adding reference to `somelib/2.3.5`? I read this like "please, gimme the library somelib with the version 2 of its public API and .3.5 version of its implementation details". From the perspective of dependency management, version is very important source of compatibility. If someone wants to use OpenGL, in fact he/she should stick to some version of this API.

Reply:
@Nipheris  I doubt this is feasible. the same implementation (DLL) of OpenGL may allow to create contexts of different versions. you usually choose the desired API version in runtime, based on what do you have available on the given machine. it may depends on drivers currently installed (nVidia / AMD / Intel / etc), and also compatibility of the GPU with given OpenGL specification. in other words, it's likely you may request OpenGL 4.5 if you're running GeForce 2080 Ti with recent drivers, but you'll probably fallback to OpenGL 1.0 if you're running on GeForce 4 MX 440.

Reply:
@SSE4 Sounds reasonable.. Nevertheless, what interface this package is going to provide? What headers will be copied to the package? Or what `include_paths` will be propagated to the dependent packages?

@ericLemanissier says about `a unique point choosing the opengl implementation`, but what API we are going to implement? Do we really have some "cross-platform OpenGL" all the packages should depend on, or we have NO such crossplatform API and we need the loaders like `glew` for this purpose?

Or maybe we should have `opengl-linux` package for ALL the Linux distros, but the different package for Windows?

Reply:
@SSE4 Thanks for the clarification! I was embarassed by the `package` method of the recipe, seems it is not needed at all.

So, to sum up the things:
- this general idea of this package is to provide cross-platform access to OpenGL API irrespective of the specific OpenGL spec and extensions (so we can make a claim that this Conan package provides access to ALL versions of OpenGL, available for the target system), and the specific version's API should be retrieved by other "loader" libraries like `GLEW`, which, in turn, will refer to this `opengl` package as a dependency;
- this package is going to populate `cpp_info` in a way the dependent package can consume OpenGL (again, irrespective of used spec version) in a unified way on all platforms (with the some exceptions like using `#include <OpenGL/gl.h>` instead of `#include <GL/gl.h>` on OSX), including the linking information. The differences between platforms are insignificant, so it makes sense to have such "cross-platform" OpenGL package.
- the implementation details like using vendor drivers vs mesa is a responsibility of package options, so we can change them down the road without modifying the source code at all.

Have I missed something?
Can we include such statements about this package somewhere in description?

Reply:
@Nipheris I've removed the package method, is was mainly leftover from the previous implementation.